### ISSUE TITLE
After building an asynchronous action, the extraReducers state is executed twice

### ISSUE THREAD

**Issue by JackpotM1103** [reduxjs/redux-toolkit#2882]
I initiate a request in the click event, the click event and the request are executed only once, but the asynchronous action in the extraReducers is executed twice, I hope someone who knows the problem is willing to tell me your valuable opinions, thank you.
Below is my code snippet.
`asyncAction`
```js
export const testRequest = createAsyncThunk("test/testRequest", arg =>
  axios
    .post(`/api/pages?root=${arg}`)
);
```
`extraReducers`
```js
extraReducers(builder) {
  builder
    .addCase(testRequest.pending, (state, action) => {
      console.log('This will print twice');
    })
    .addCase(testReuqest.fulfilled, (state, action) => {
      console.log('This will print twice');
    });
}
```

**Comment by markerikson** [comment]
Where are you actually dispatching that thunk? Can you put together a CodeSandbox or a Replay 9 ( https://replay.io/record-bugs ) that shows what you're doing?

**Comment by JackpotM1103** [comment]
@markerikson Due to the strict confidentiality of the enterprise and the webview-based development model, I'm sorry I can't share my code on codeSandbox or Replay 9, but after debugging and console, I'm sure that the click event and api request are executed only once, strangely the asyncAction's Both requestIds are exactly the same.
![image](https://user-images.githubusercontent.com/86994283/200729272-ec02310c-c885-4c48-8431-798c5c3907b9.png)
If possible, can you tell me under what circumstances requestId can be repeated twice? Because I don`t know how this happened

**Comment by markerikson** [comment]
@JackpotM1103 : unfortunately, that really _shouldn't_ be happening, ever.  As in, it shouldn't be _possible_ for that to happen without the entire thunk being dispatched twice.  That's why I was asking for a repro or a replay so I can dig into _why_ that's happening.

My only immediate suggestion would be to add some `console.trace()` calls  to that reducer.

Actually, if you've got the Redux DevTools available in some way with that webview, the "Trace" tab ought to give you a stack trace for where that dispatch is coming from, although it's probably just going to point into `createAsyncThunk` in general.

Another thought would be to edit the RTK code in `node_modules` to add logging inside of the thunk and see if it is running more than once.

**Comment by phryneas** [comment]
Do you maybe have a custom middleware in there that executes `next` twice?

**Comment by JackpotM1103** [comment]
@markerikson: Thanks for your suggestion, I used redux-logger library to print redux call logs. It looks fine on the log. The pending and fulfilled states of async getColumns only fire once, but the code inside the pending and fulfilled functions still executes twice.
![image](https://user-images.githubusercontent.com/86994283/200767279-7789db04-6e4a-4df7-8632-9cd47235204b.png)

**Comment by JackpotM1103** [comment]
> Do you maybe have a custom middleware in there that executes `next` twice?

I'm sure I don't use any other middleware, just use dispatch to trigger asyncAction, and then listen for pending and fulfilled in extraReducers, and through redux-logger the above statement is determined

**Comment by phryneas** [comment]
Well, redux-logger is a middleware, so you have at least some kind of middleware set up now. Could you please share your store setup?

Also, did you maybe mount your slice twice?

**Comment by JackpotM1103** [comment]
@phryneas 
What does it mean to mount a slice twice? Is it createSlice or something else ?

Here is my directory structure

```shell
- src
  -- store
    --- index.js
    --- silverReducers.js
  -- components
    --- Tree
      ---- index.js
      ---- treeSlice.js
    --- Table
      ---- index.js
      ---- tableSlice.js
```

The following is my store related settings

`src/store/index.js`

```js
import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import silverReducer from "./silverAntReducers";
import treeReducer from "@components/Tree/treeSlice";
import tableReducer from "@components/Table/tableSlice";

export default configureStore({
  reducer: {
    silver: silverReducer,
    tree: treeReducer,
    table: tableReducer,
  },
  devTools: process.env.NODE_ENV !== "production",
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(logger),
});
```

**Comment by phryneas** [comment]
> What does it mean to mount a slice twice?

It would be something like
```js
export default configureStore({
  reducer: {
    silver: silverReducer,
    tree: treeReducer,
    table: tableReducer,
    somethingElse: tableReducer // another `tableReducer`
  },
  devTools: process.env.NODE_ENV !== "production",
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(logger),
});
```

**Comment by JackpotM1103** [comment]
> > What does it mean to mount a slice twice?
> 
> It would be something like

Thank you very much for your valuable comments! It is precisely because the slice is installed twice in the store that the code in the fulfilled execution body in the tableReducer is executed twice, so this problem has been solved

**Comment by phryneas** [comment]
Is there any reason why you are doing this though? Those two slices will always have exactly the same value, so it seems very redundant.

**Comment by JackpotM1103** [comment]
> Is there any reason why you are doing this though? Those two slices will always have exactly the same value, so it seems very redundant.

Yes, because I have a requirement to merge two projects that have reusable Table components, I am using redux and have not noticed the problem of duplicating the installation of slices
